src/SigninRequest.ts

```diff
@@ -104,7 +104,7 @@ export class SigninRequest {
         }
 
         parsedUrl.searchParams.append("state", this.state.id);
-        if (this.state.code_challenge) {
+        if (this.state.code_challenge && !optionalParams.disablePKCE) {
```

Comment:
`&& !optionalParams.disablePKCE` should not be needed, as `this.state.code_challenge` should be undefined, because of adapted `SigninState` above